backend/uclapi/roombookings/decorators.py

```diff
@@ -55,6 +114,31 @@ def wrapped(request, *args, **kwargs):
 
         queryparams = dict(request.GET)
 
+        token = request.GET["token"]
+
+        try:
+            if token.split("-")[1] == "temp":
+                is_temp_token = True
+        except IndexError:
+            is_temp_token = False
+
+        user = None
+
+        if is_temp_token:
```

Comment:
L125-L126 might never get called since you check for that already in `does_token_exist` which means at this point you already checked if it exists & possibly throttled it so here you either have a valid {temp, permanent} token